data-pipeline/src/pipeline/part_year/common.py

```diff
@@ -0,0 +1,51 @@
+import pandas as pd
+
+from src.pipeline import config
+
+
+def map_has_pupil_comparator_data(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Whether the data is sufficiently populated to create a pupil
+    comparator group.
+
+    Specifically, this is the FSM and SEN data.
+
+    :param df: academy/school data
+    :return: updated DataFrame
+    """
+    pupil_comparator_columns = list(
+        set(list(config.census_column_map.values()) + config.sen_generated_columns)
+    )
+
+    df["Pupil Comparator Data Present"] = (
+        ~df[pupil_comparator_columns].isna().all(axis=1)
+    )
+
+    return df
+
+
+def map_has_building_comparator_data(
+    maintained_schools: pd.DataFrame,
+) -> pd.DataFrame:
+    """
+    Whether the maintained school data has all the necessary data to
+    create a building comparator group.
+
+    Specifically, this is the CDC data.
+
+    Note: "Total Internal Floor Area" is specifically excluded as this
+    value is previously set to the _median_ if absent.
+
+    :param maintained_schools: maintained schools data
+    :return: updated DataFrame
+    """
+    building_comparator_columns = [
+        column for column in config.cdc_generated_columns
+        if column != "Total Internal Floor Area"
```

Comment:
As documented in the docstring, this is a workaround as this gets a default value in `build_academy_data()` if it's NULL.

Thinking about it, could that actually be removed as it later gets the same default value in `prepare_data()` before comparator-sets are determined. Need to figure out if it's needed before that.

Reply:
Note: this can't be removed as per the above comment as that default value is required for apportionment.